tests/utils/test_url.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+from page_analyzer.utils.url import (
+    check_url,
+    normalize,
+    parse_page_data,
+    validate,
+)
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {}
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    expected_errors = {"name": "Некорректный URL"}
+    assert errors == expected_errors
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    expected_url = "http://www.example.com"
+    assert normalized_url == expected_url
+
+
+class TestCheckUrl(unittest.TestCase):
+    @patch("page_analyzer.utils.url.parse_page_data")
```

Comment:
wdyt about using [patch](https://docs.python.org/3/library/unittest.mock.html#quick-guide) as a context manager?
<img width="825" alt="image" src="https://github.com/AlexMomot-717/python-project-83/assets/45393914/325f599e-be4e-4b9c-999c-20e4dc124aa7">

it looks better in the `given - when - then` tests setup IMO
and don't forget to check that mocks were called!